### ISSUE TITLE
can't reproduce mnist result

### ISSUE THREAD

**Issue by xuyujie2** [patrickschlachter/osr-using-ics#1]
hi, i am trying to reproduce the result of baccu that you put on the paper.  unfortunately, i only get 0.8859 using model named 'modified_vgg' on mnist dataset,while it is 0.943 on the paper.i dont know if i should change model or something .the setting is same as the github project but only change the default dataset with mnist.

**Comment by patrickschlachter** [comment]
Hi,
thank you for your interest. You use the right model. However, the performance depends on the choice of known classes set by the random seed. Moreover, the hyperparameters in the code were set for CIFAR-10. Regarding MNIST, e.g. the number of training epochs for both splitting and pretraining should be less than 25. If you're interested, we can give you further details on hyperparameters next week.

Best regards,
Patrick

**Comment by xuyujie2** [comment]
thanks for reply. i will try to change the training class set and some hyperpara like you said.it would be nice if more  details are told.very thanks again!

**Comment by xuyujie2** [comment]
Hi, there is another question: when splitting,  the code shows splitting on the whole dataset. In that case, it may occurs that most of splitting data belong to one class which performs poorer than others. Is it more reasonable that we take  same number atynormal samples from each class?

**Comment by xuyujie2** [comment]
Hi, you are right! I change the training set of mnist to [0,2,3,5,6,8], and set splitting and pretrain both 25 epoch . I got baccu = 0.9429.

**Comment by patrickschlachter** [comment]
> Hi, you are right! I change the training set of mnist to [0,2,3,5,6,8], and set splitting and pretrain both 25 epoch . I got baccu = 0.9429.

Great that it works fine now!

> Hi, there is another question: when splitting, the code shows splitting on the whole dataset. In that case, it may occurs that most of splitting data belong to one class which performs poorer than others. Is it more reasonable that we take same number atynormal samples from each class?

Yes, defining the same proportion of samples per class as atypical makes sense. This _class-balanced splitting_ is especially useful, when the distributions and amounts of samples between the known classes differ significantly. Regarding the examined datasets, we conducted experiments with class-balanced splitting some time ago and it lead to results comparable to those in the paper.

**Comment by xuyujie2** [comment]
Hi, i only got baccu=0.6522 of cifar_10 using default setting. What should i change?

**Comment by patrickschlachter** [comment]
The default settings should work for CIFAR-10. Did you again try different combinations of known classes set by the random seed?

**Comment by qwbljs** [comment]
hello, i have the same question. Default order [1, 3, 4, 6, 7, 9] of cifar10 did not work for me. i only get baccu=0.6051. The other setting is same as this project.     
    RHO = 10
    LOSS_WEIGHTS = [1., 1.]
    SPLIT_METHOD = 'cnn'
    TRAIN_EPOCH = 100
    PRE_TRAIN_EPOCH = 50
best confusion matrix is :
[[1171  284  636  708  217  256  728]
 [  29  885    2    0    4    2   78]
 [ 248    5  609   60   39   24   15]
 [ 103    3   36  810   23   25    0]
 [ 112    6   50   37  778    7   10]
 [  70    3   18   60    6  831   12]
 [  54   31    4    3    1    5  902]]
